libcore/Interfaces/SidebarListInterface.vala

```diff
@@ -1,69 +1,46 @@
 /*
- * SidebarListInterface.vala
- *
- * Copyright 2020 elementary LLC. <https://elementary.io>
- *
- * This program is free software; you can redistribute it and/or modify
- * it under the terms of the GNU General Public License as published by
- * the Free Software Foundation; either version 2 of the License, or
- * (at your option) any later version.
- *
- * This program is distributed in the hope that it will be useful,
- * but WITHOUT ANY WARRANTY; without even the implied warranty of
- * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
- * GNU General Public License for more details.
- *
- * You should have received a copy of the GNU General Public License
- * along with this program; if not, write to the Free Software
- * Foundation, Inc., 51 Franklin Street, Fifth Floor, Boston,
- * MA 02110-1301, USA.
+ * SPDX-License-Identifier: GPL-2.0+
+ * SPDX-FileCopyrightText: 2020-2023 elementary, Inc. (https://elementary.io)
  *
  * Authors : Jeremy Wootten <jeremy@elementaryos.org>
  */
 
-public interface Sidebar.SidebarListInterface : Gtk.Container {
+public interface Sidebar.SidebarListInterface : Object {
     public abstract Files.SidebarInterface sidebar { get; construct; }
+    public abstract Gtk.ListBox list_box { get; internal set; }
 
-    public abstract void select_item (SidebarItemInterface? item);
+    public abstract void select_item (Gtk.ListBoxRow? item);
     public abstract void unselect_all_items ();
 
-    public virtual void open_item (SidebarItemInterface item, Files.OpenFlag flag = Files.OpenFlag.DEFAULT) {
+    public virtual void open_item (SidebarItemInterface item, Files.OpenFlag flag = DEFAULT) {
         sidebar.path_change_request (item.uri, flag);
     }
 
     public abstract void refresh (); //Clear and recreate all rows
     public virtual void refresh_info () {} //Update all rows without recreating them
 
-    public virtual uint32 add_plugin_item (Files.SidebarPluginItem plugin_item) {return 0;}
+    public virtual uint32 add_plugin_item (Files.SidebarPluginItem plugin_item) {
+        return 0;
+    }
 
     public virtual void clear () {
-        foreach (Gtk.Widget child in get_children ()) {
-            remove (child);
+        foreach (unowned var child in list_box.get_children ()) {
             if (child is SidebarItemInterface) {
-                ((SidebarItemInterface)child).destroy_bookmark ();
+                ((SidebarItemInterface) child).destroy_bookmark ();
             }
+
+            list_box.remove (child);
```

Comment:
I don't fully understand this, but if we call `remove` before `destroy` we get a whole bunch of errors about how this isn't a widget or isn't a container and there's unclassed pointers etc etc. But if we destroy and then remove there's nothing :woman_shrugging:

Reply:
It is because you changed `Gtk.Widget` to  `unowned var `.  If you remove `child` from the listbox there is nothing referencing it so you cannot safely refer to it again.  Using `Gtk.Widget` adds another reference so the order of subsequent actions does not matter.

Reply:
Ah I had no idea that the `var` keyword could have an effect on references. Thanks!